.travis.yml

```diff
@@ -91,78 +159,94 @@ matrix:
       after_success:
         - ./platform/node/scripts/after_success.sh
 
-    # GCC 5 - Debug - Coverage
+    # OSMesa - GCC 5 - Debug (Coverage)
     # FIXME: https://github.com/mapbox/mapbox-gl-native/issues/6918
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
-      compiler: "glfw-gcc5-debug"
+      compiler: "osmesa-gcc5-debug"
       env: BUILDTYPE=Debug _CXX=g++-5 _CC=gcc-5 WITH_COVERAGE=1 WITH_OSMESA=1
       addons: *gcc5
       after_script:
         - ccache --show-stats
         - ./platform/linux/scripts/coveralls.sh
 
-    # GCC 5 - Release
+    # OSMesa - GCC 5 - Release
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
-      compiler: "glfw-gcc5-release"
+      compiler: "osmesa-gcc5-release"
       env: BUILDTYPE=Release _CXX=g++-5 _CC=gcc-5 WITH_OSMESA=1
       addons: *gcc5
 
-    # Clang 3.8 - Debug
+    # OSMesa - Clang 3.8 - Debug
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
-      compiler: "glfw-clang38-debug"
+      compiler: "osmesa-clang38-debug"
       env: BUILDTYPE=Debug _CXX=clang++-3.8 _CC=clang-3.8 WITH_OSMESA=1
       addons: *clang38
 
-    # Clang 3.8 - Release
+    # OSMesa - Clang 3.8 - Release
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
-      compiler: "glfw-clang38-release"
+      compiler: "osmesa-clang38-release"
       env: BUILDTYPE=Release _CXX=clang++-3.8 _CC=clang-3.8 WITH_OSMESA=1
       addons: *clang38
 
-    # Clang 3.8 - check
+    # EGL - GCC 5 - Release
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
-      compiler: "check-clang38-release"
-      env: BUILDTYPE=Release _CXX=clang++-3.8 _CC=clang-3.8 WITH_OSMESA=1
-      addons: *clang38
-      script:
-        - git fetch origin master:refs/remotes/origin/master
-        - make check
+      compiler: "egl-gcc5-release"
+      env: BUILDTYPE=Release _CXX=g++-5 _CC=gcc-5 WITH_EGL=1
+      addons: *gcc5
+      before_script:
+        - mapbox_start_xvfb
 
-    # Qt 4 - Release
+    # GLX - GCC 5 - Release
+    - os: linux
+      sudo: required
+      dist: trusty
+      language: cpp
+      compiler: "glx-gcc5-release"
+      env: BUILDTYPE=Release _CXX=g++-5 _CC=gcc-5
+      addons: *gcc5
+      before_script:
+        - mapbox_start_xvfb
+
+    # Qt 4 - GCC 5 - Release
     - os: linux
       sudo: required
       dist: trusty
       language: cpp
       compiler: "qt4-gcc5-release"
-      env: BUILDTYPE=Release _CXX=g++-5 _CC=gcc-5 RUN_XVFB=1 WITH_QT_4=1
+      env: BUILDTYPE=Release _CXX=g++-5 _CC=gcc-5 WITH_QT_4=1
       addons: *qt4
+      before_script:
```

Comment:
Should the `GLX - Node - Clang 3.8 - Release` build currently using `RUN_XVFB` be switched to these setup scripts instead?